src/org/javarosa/core/model/ItemsetBinding.java

```diff
@@ -114,32 +114,34 @@ public IConditionExpr getRelativeValue () {
     public void initReferences(QuestionDef q) {
         // To construct the xxxRef, we need the full model, which wasn't available before now.
         // Compute the xxxRefs now.
-        nodesetRef = FormInstance.unpackReference(FormDef.getAbsRef(new XPathReference(
-                ((XPathPathExpr) ((XPathConditional) nodesetExpr).getExpr()).getReference()), contextRef));
-        if ( labelExpr != null ) {
-            labelRef = FormInstance.unpackReference(FormDef.getAbsRef(new XPathReference(((XPathPathExpr) ((XPathConditional) labelExpr).getExpr())),
-                    nodesetRef));
-        }
-        if ( copyExpr != null ) {
-            copyRef = FormInstance.unpackReference(FormDef.getAbsRef(new XPathReference(((XPathPathExpr) ((XPathConditional) copyExpr).getExpr())),
-                    nodesetRef));
-        }
-        if ( valueExpr != null ) {
-            valueRef = FormInstance.unpackReference(FormDef.getAbsRef(new XPathReference(((XPathPathExpr) ((XPathConditional) valueExpr).getExpr())),
-                    nodesetRef));
-        }
 
-        if ( q != null ) {
+        // Convert the expression to a relative reference and then anchor it to the context.
+        nodesetRef = (TreeReference) getAbsRef(new XPathReference(
```

Comment:
You should be able to just use `getAbsoluteRef(nodesetExpr, contextRef)`, here. Did that not work?